src/include/homeobject/common.hpp

```diff
@@ -25,11 +25,26 @@ constexpr uint64_t Gi = Ki * Mi;
 
 namespace homeobject {
 
-using blob_id = uint64_t;
+using blob_id_t = uint64_t;
 
-using peer_id = boost::uuids::uuid;
-using pg_id = uint16_t;
-using shard_id = uint64_t;
+using peer_id_t = boost::uuids::uuid;
+using pg_id_t = uint16_t;
+using shard_id_t = uint64_t;
+
+template < typename T >
+using shared = std::shared_ptr< T >;
+
+template < typename T >
+using cshared = const std::shared_ptr< T >;
+
+template < typename T >
+using unique = std::unique_ptr< T >;
+
+template < typename T >
+using intrusive = boost::intrusive_ptr< T >;
+
+template < typename T >
+using cintrusive = const boost::intrusive_ptr< T >;
```

Comment:
I don't think any of these `pointer` types are used in the API; can we move them to `homeobject_impl.hpp` (or create a new common_impl.hpp there?)

Reply:
We certainly use it HO, but agree we can move this to homeobject_impl.hpp

Reply:
But isn't it better be consistent across repo and I don't see anything wrong declaring here either, if homeobject chose to return on of these. So I am very much tempted to keep it here itself.

Reply:
If we choose to return them we lift them here? I'm not sure I understand the use of providing a `homeobject::shared` to the consumer when they can use `std::shared_ptr` which is arguably shorter and more known?

Reply:
Moved to homeobject_impl.hpp